xdev_bot/issues.py

```diff
@@ -8,79 +8,74 @@
-    - construct URL for card
-
-    1. Whenever an issue is closed:
-    - get column ID of "done" column
-    - retrieve updated_at from payload and update database
-    - update card information (new column ID)
-
-    2. Whenever an issue is reopened:
-    - get column ID of "in-progess" column
-    - update card information (new column ID)
+    else:
+        df = read_database()
+        row = df['note'] == html_url
+        card_id = df.loc[row]['card_id'].values[0]
 
+        if event.data['action'] == 'closed':
+            card_action = 'Closing'
+            column_id = PROJECT_BOARD['columns']['done']['id']
 
-    """
-    issue_url = event.data['issue']['html_url']
-    project_board_name = PROJECT_BOARD['name']
-    df = read_database()
-    row = df['note'] == issue_url
-    card_id = df.loc[row]['card_id'].values[0]
-    url = f'/projects/columns/cards/{card_id}/moves'
+            if event.data[0] == 'pull_request':
+                dict_of_labels = event.data['pull_request']['labels']
+                labels = []
 
-    if event.data['action'] == 'closed':
-        done_column_id = PROJECT_BOARD['columns']['done']['id']
-        print(f'Updating database: move card {card_id} to done column')
+                for item in dict_of_labels: labels.append(item['name'])
+                labels = set(labels)
 
-        print(f'Closing Card in {project_board_name} project board for issue : {issue_url}')
-        # POST /projects/columns/cards/:card_id/moves
+                if 'needs-review' in labels: labels.remove('needs-review')
 
-        await gh.post(
-            url,
-            data={'position': 'top', 'column_id': done_column_id},
-            accept='application/vnd.github.inertia-preview+json',
-        )
+                merged = event.data['pull_request']['merged']
+                if merged: labels.add('merged')
+                else: labels.add('rejected')
+        
+                labels = list(labels)
+                await gh.patch(issue_url, data={'labels': labels})
 
-    else:
-        in_progress_column_id = PROJECT_BOARD['columns']['in_progress']['id']
-        print(f'Updating database: move card {card_id} to in-progress column')
+        elif event.data['action'] == 'reopened':
+            card_action = 'Reopening'
+            column_id = PROJECT_BOARD['columns']['in_progress']['id']
 
-        await gh.post(
-            url,
-            data={'position': 'top', 'column_id': in_progress_column_id},
-            accept='application/vnd.github.inertia-preview+json',
-        )
-        print(f' Reopening card in {project_board_name} project board for issue : {issue_url}')
+        url = f'/projects/columns/cards/{card_id}/moves'
 
+    await gh.post(
+        url,
+        data={'position': 'top', 'column_id': column_id},
+        accept='application/vnd.github.inertia-preview+json')
+    print(f' {card_action} card in {project_board_name} project board for issue : {html_url}')
 
```

Comment:
All of the labeling stuff can come at the end, here, with a single `gh.patch` call.